@xen-orchestra/lite/src/components/AppLogin.vue

```diff
@@ -69,15 +69,26 @@ watch(password, () => {
 async function handleSubmit() {
   try {
     await xenApiStore.connect(login.value, password.value);
-  } catch (err) {
+  } catch (err: any) {
     if ((err as Error).message === "SESSION_AUTHENTICATION_FAILED") {
       focusPasswordInput();
       isInvalidPassword.value = true;
       error.value = t("password-invalid");
-    } else {
-      error.value = t("error-occurred");
-      console.error(err);
+      return;
     }
+
+    if (err.message === "HOST_IS_SLAVE") {
+      const [masterIp] = err.data;
+      const url = new URL(window.origin);
+      url.hostname = masterIp;
+      error.value = t("cannot-login-from-slave", {
+        masterLink: `<a href="${url.href}">${t("go-to-master")}</a>`,
```

Comment:
IMO, we shouldn't use raw HTML like this.

Could we imagine a `LoginError` component which would take the error as prop?

```vue
<LoginError :error="error" />
```

It'd be up to it to handle the error display.

```vue
<div class="error" v-if="error !== undefined">
  <span v-if="error.message === 'SESSION_AUTHENTICATION_FAILED'">
    {{ $t("password-invalid") }} 
  </span>
  <span v-else-if="error.message === 'HOST_IS_SLAVE'">
    {{ $t("cannot-login-from-slave") }} 
    <a href="...">${t("go-to-master")}</a>
  </span>
  <span v-else>
    {{ $t("error-occurred") }}
  </span>
</div>
```